src/pages/Index.tsx

```
@@ -1,210 +1,243 @@
+
+  // Notification WebSocket useEffect (as before)
+  useEffect(() => {
+    const wsProtocol = window.location.protocol === 'https:' ? 'wss' : 'ws';
+    const backendHost = import.meta.env.VITE_API_BASE_URL?.replace(/^https?:\/\//, '') || 'localhost:8000';
+    const wsUrl = `${wsProtocol}://${backendHost}/ws/notifications`;
+    notificationWs.current = new WebSocket(wsUrl);
+    notificationWs.current.onopen = () => { console.log('Notification WS connected.'); toast({ title: "System", description: "Connected to live updates.", duration: 2000 }); };
+    notificationWs.current.onmessage = (event) => {
+      try {
+        const rawMessage = JSON.parse(event.data as string);
+        if (rawMessage && rawMessage.chat_room_id && rawMessage.request_id) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)